File: 4/go.js
var fs = require("fs");

function part1(err, data){
    const cnt = data.split("\n").reduce((acc, cur) => {
        const p1 = cur.split(",")[0], p2 = cur.split(",")[1]
        if (p2 == undefined) return acc;
        const l1 = parseInt(p1.split("-")[0]), r1 = parseInt(p1.split("-")[1]);
        const l2 = parseInt(p2.split("-")[0]), r2 = parseInt(p2.split("-")[1]);
        if ((l1 <= l2 && r1>=r2) || (l1>=l2 && r1<=r2))
            return acc + 1
        else return acc
    }, 0);
    console.log("Final sol of part 1: ", cnt);
}

function part2(err, data){
    const cnt = data.split("\n").reduce((acc, cur) => {
        const p1 = cur.split(",")[0], p2 = cur.split(",")[1]
        if (p2 == undefined) return acc;
        const l1 = parseInt(p1.split("-")[0]), r1 = parseInt(p1.split("-")[1]);
        const l2 = parseInt(p2.split("-")[0]), r2 = parseInt(p2.split("-")[1]);
        if ((r1<l2) || (r2<l1))
            return acc
        else return acc + 1
    }, 0);
    console.log("Final sol of part 2: ", cnt);
}


// fs.readFile('input.txt', 'utf8', (err, data) => part1(err, data));
fs.readFile('input.txt', 'utf8', (err, data) => part2(err, data));

File: 7/go.go
package main

import (
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	// content, err := os.ReadFile("smol.txt")
	content, err := os.ReadFile("input.txt")
	sizes := make(map[int]int)
	if err != nil {
		log.Fatal(err)
	}
	lines := strings.Split(string(content), "\n")
	inputs := []string{}
	for _, line := range lines {
		nl := strings.Replace(line, "$ ", "", 1)
		if nl[0:2] != "ls" && nl[0:3] != "dir" {
			inputs = append(inputs, nl)
		}
	}

	stack := []int{}
	for i := 0; i < len(inputs); i++ {
		command := inputs[i]
		if command[0:2] == "cd" && strings.Contains(command, "..") {
			stack = stack[:len(stack)-1]
		} else if command[0:2] == "cd" {
			stack = append(stack, i)
			sizes[i] = 0
		} else {
			size, _ := strconv.Atoi(strings.Split(string(command), " ")[0])
			for _, s := range stack {
				sizes[s] += size
			}
		}
	}
	ans_1 := 0
	for _, size := range sizes {
		if size <= 100000 {
			ans_1 += size
		}
	}

	ans_2 := 70000000
	n_space := 30000000 - (70000000 - sizes[0])

	for _, size := range sizes {
		if size >= n_space && size <= ans_2 {
			ans_2 = size
		}
	}

	fmt.Println("sol for part 1: ", ans_1)
	fmt.Println("sol for part 2: ", ans_2)
}
